python/Ganga/GPIDev/Base/Objects.py

```diff
@@ -74,6 +73,7 @@ class Node(object):
     thread-safe usage.
     """
     __metaclass__ = abc.ABCMeta
+    __slots__ = ('_index_cache_dict', '_parent', '_read_lock', '_write_lock', '_dirty', '_registry', '_data_dict')
```

Comment:
For the whole branch several hundred Mb, for slots I'll let you know once I've some more commits put in addressing some other stuff then I'll run with and without.